projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Invoices.php

```diff
@@ -1794,7 +1799,20 @@ public function addUpdateInvoice($args=array()){
                         }
 
         } else {
-            
+					// If we are using our CRM reference id (table field id_override) system, we should generate a new invoice number.
+					if ( isset( $data['woo_use_crm_id'] ) && $data['woo_use_crm_id'] === true ) {
+						$ref_type = $zbs->settings->get( 'reftype' );
+						// We can only generate it if autonumber is set
+						if ( $ref_type === 'autonumber' ) {
+							$next_number                 = $zbs->settings->get( 'refnextnum' );
+							$prefix                      = $zbs->settings->get( 'refprefix' );
+							$suffix                      = $zbs->settings->get( 'refsuffix' );
+							$dataArr['zbsi_id_override'] = $prefix . $next_number . $suffix; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
```

Comment:
```suggestion
							$next_number                 = $zbs->settings->get( 'refnextnum' );
							$dataArr['zbsi_id_override'] = $zbs->settings->get( 'refprefix' ) . $next_number . $zbs->settings->get( 'refsuffix' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
```

Could we remove two of those variables as we don't use them anywhere else?